### PR TITLE
Fix `scroll_offset_changed` signal emitting every time graph is re-drawn

### PR DESCRIPTION
**Pull request by ParanoidHawk**
This fix ensures `GraphEdit`'s `scroll_offset_changed` signal is fired only when the scroll values are actually changed.
Contains a small related fix to prevent regression of the issue with `AnimationNodeBlendTreeEditor` resetting scroll view when a connection is made.

*Bugsquad edit:* Fixes #76270

### CODE REVIEW

**Comment by KoBeWi** [comment]
Note that we have `set_value_no_signal()` method now, which sets the value without emitting the signal. I think `setting_scroll_ofs` could be removed entirely now (perhaps).

File: editor/plugins/animation_blend_tree_editor_plugin.cpp [hunk 1]
```diff
@@ -125,8 +125,6 @@ void AnimationNodeBlendTreeEditor::update_graph() {
 
 	visible_properties.clear();
 
-	graph->set_scroll_ofs(blend_tree->get_graph_offset() * EDSCALE);
```
Comment (YuriSizov) [inline editor/plugins/animation_blend_tree_editor_plugin.cpp:128]: I'm pretty sure this is required to center the view on the blend tree.

File: scene/gui/graph_edit.h [hunk 2]
```diff
@@ -345,6 +345,7 @@ class GraphEdit : public Control {
 	void remove_valid_left_disconnect_type(int p_type);
 
 	void set_scroll_ofs(const Vector2 &p_ofs);
+	void _set_scroll_ofs(const Vector2 &p_ofs);
```
Comment (YuriSizov) [inline scene/gui/graph_edit.h:348]: If the added method is supposed to be private, it needs to be put into the appropriate section above.

File: scene/gui/graph_edit.cpp [hunk 3]
```diff
@@ -260,7 +275,13 @@ void GraphEdit::_scroll_moved(double) {
 	if (!awaiting_scroll_offset_update) {
 		call_deferred(SNAME("_update_scroll_offset"));
 		awaiting_scroll_offset_update = true;
+
+		// In Godot, signals on change value are avoided as a convention.
+		if (!setting_scroll_ofs) {
+			emit_signal(SNAME("scroll_offset_changed"), get_scroll_ofs());
+		}
```
Comment (YuriSizov) [inline scene/gui/graph_edit.cpp:282]: Moving it here means that we will emit the signal before the value has been updated, which is incorrect.

File: scene/gui/graph_edit.cpp [hunk 4]
```diff
@@ -246,10 +246,25 @@ void GraphEdit::get_connection_list(List<Connection> *r_connections) const {
...
 	setting_scroll_ofs = true;
+	_set_scroll_ofs(p_ofs);
+	setting_scroll_ofs = false;
+}
+
+void GraphEdit::_set_scroll_ofs(const Vector2 &p_ofs) {
+	bool is_being_set = setting_scroll_ofs;
+	// _scroll_moved is called when each scroll value is set.
+	// So suppress the scroll_offset_changed signal until after they're set.
+	setting_scroll_ofs = true;
```
Comment (YuriSizov) [inline scene/gui/graph_edit.cpp:257]: This double use of the same flag is too convoluted and unreliable. I think the overall solution can be as simple as storing the the last offset and checking if it has changed before emitting the signal.

**Pull request closed without merging**